larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.cc

```diff
@@ -0,0 +1,164 @@
+
+void TwoViewTransverseTracksValidationTool::CollectTruthInformation(const Cluster *const pCluster, TreeDataBox &treeDataBox, std::string clusterName)
+{
+    constexpr float floatDefault(std::numeric_limits<int>::lowest());
+
+    int foundMCParticle(0);
+    int pdgMCParticle(std::numeric_limits<int>::lowest());
+    int isPrimaryMCParticle(0);
+    float energyMCParticle(floatDefault);
+    float momXMCParticle(floatDefault), momYMCParticle(floatDefault), momZMCParticle(floatDefault);
+    float posXMCParticle(floatDefault), posYMCParticle(floatDefault), posZMCParticle(floatDefault);
+
+    const MCParticle *const pMCParticle(MCParticleHelper::GetMainMCParticle(pCluster));
+    if (pMCParticle)
+    {
+        foundMCParticle = 1;
+        pdgMCParticle = pMCParticle->GetParticleId();
+        isPrimaryMCParticle = static_cast<int>(pMCParticle->IsRootParticle());
+        energyMCParticle = pMCParticle->GetEnergy();
+        momXMCParticle = pMCParticle->GetMomentum().GetX();
+        momYMCParticle = pMCParticle->GetMomentum().GetY();
+        momZMCParticle = pMCParticle->GetMomentum().GetZ();
+        posXMCParticle = pMCParticle->GetVertex().GetX();
+        posYMCParticle = pMCParticle->GetVertex().GetY();
+        posZMCParticle = pMCParticle->GetVertex().GetZ();
+    }
+
+    StoreAndRegisterDatum(foundMCParticle, ("foundMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(pdgMCParticle, ("pdgMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(isPrimaryMCParticle, ("isPrimaryMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(energyMCParticle, ("energyMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momXMCParticle, ("momXMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momYMCParticle, ("momYMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momZMCParticle, ("momZMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posXMCParticle, ("posXMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posYMCParticle, ("posYMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posZMCParticle, ("posZMCParticle" + clusterName), treeDataBox);
+
+    return;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewTransverseTracksValidationTool::CollectChargeProfileInformation(
+    const DiscreteProbabilityVector &discreteProbabilityVector, TreeDataBox &treeDataBox, std::string clusterName)
+{
+    std::string xName("xValues" + clusterName);
+    std::string probabilityDensityName("probDensityValues" + clusterName);
+    std::string probabilityName("probValues" + clusterName);
+    std::string cumulativeProbabilityName("cumulProbValues" + clusterName);
+    std::string widthName("widthValues" + clusterName);
```

Comment:
Do const strings work here, or does it cause issues with the `TreeDataBox`?

Reply:
No issues caused, just a slip up on my end.
TY @lhwhitehead